### PR TITLE
VSTS-301 Fix semver check for compatibility with SQ >= 10.0

### PR DESCRIPTION
**Pull request by mickael-caro-sonarsource**
Please review our [contribution guidelines](https://github.com/SonarSource/sonar-scanner-vsts/blob/master/docs/contributing.md).

And please ensure your pull request adheres to the following guidelines: 

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](https://jira.sonarsource.com/browse/VSTS) ticket available, please make your commits and pull request start with the ticket ID (VSTS-XXXX)

### CODE REVIEW

**Comment by alexanderameye** [comment]
As a temporary workaround, this worked for me.

```
- task: CopyFiles@2
   inputs:
     SourceFolder: $(Agent.TempDirectory)
     Contents: '**/report-task.txt'
     TargetFolder: $(Agent.BuildDirectory)   
```

**Comment by jaz2038** [comment]
@mickael-caro-sonarsource do you know if a merge here is likely soon?

File: commonv5/ts/sonarqube/__tests__/TaskReport-test.ts [hunk 1]
```diff
@@ -135,6 +135,62 @@ it("should find report files for SonarQube above 7.2.0", async () => {
...
+  // path separators in Windows/non-Windows
+  const expectedSearchPath = path.join(
+    "sonar",
+    tl.getVariable("Build.BuildNumber"),
+    "**",
+    "report-task.txt"
+  );
+  expect(tl.findMatch).toHaveBeenCalledTimes(1);
+  expect(tl.findMatch).toHaveBeenCalledWith("mock root search path", expectedSearchPath);
+});
```
Comment (wouter-admiraal-sonarsource) [inline commonv5/ts/sonarqube/__tests__/TaskReport-test.ts:192]: Why this test? What value does it bring compared to the identical one above? ("should find report files for SonarQube above 10.0")

**Review by wouter-admiraal-sonarsource — APPROVED** [review]
LGTM 👍 . I just have one doubt regarding the new tests.

File: commonv5/ts/sonarqube/__tests__/TaskReport-test.ts (hunk 1 above)
Comment (mickael-caro-sonarsource) [reply commonv5/ts/sonarqube/__tests__/TaskReport-test.ts]: Yes the idea was to test few other major version to make sure we will be ok. Thanks for the tip, I applied it.

**Pull request merged**